t/14-isotovideo.t

```diff
@@ -238,7 +246,59 @@ subtest 'productdir variable relative/absolute' => sub {
     unlike $log, qr/assert_screen_fail_test/, 'assert screen test not scheduled';
 };
 
+subtest 'exit status from test results: no test scheduled' => sub {
+    # dummy isotovideo invocation
+    chdir($pool_dir);
+    path(bmwqemu::STATE_FILE)->remove if -e bmwqemu::STATE_FILE;
+    path('vars.json')->remove if -e 'vars.json';
+    # TODO: backend/baseclass.pm raises if serial0 does not exist.
+    path('serial0')->touch();
+    my $module_basename = 'failing_module';
+    my $module = "tests/$module_basename";
+    my $log = combined_from { isotovideo(
+            default_opts => '--exit-status-from-test-results backend=null',
+            opts => "casedir=$data_dir/tests schedule=$module", exit_code => 101) };
+
+    like $log, qr/scheduling $module_basename $module\.pm/, 'module scheduled';
+    like $log, qr/Test result \[testresults\/result\-$module_basename\.json\] fail/, 'module test result parsed';
+};
+
+subtest 'exit status from test results: tests scheduled' => sub {
+    # dummy isotovideo invocation
+    chdir($pool_dir);
+    path(bmwqemu::STATE_FILE)->remove if -e bmwqemu::STATE_FILE;
+    path('vars.json')->remove if -e 'vars.json';
+    # TODO: backend/baseclass.pm raises if serial0 does not exist.
+    path('serial0')->touch();
+    my $module_basename = 'failing_module';
+    my $module = "tests/$module_basename";
+    my $log = combined_from { isotovideo(
+            default_opts => '--exit-status-from-test-results backend=null',
+            opts => "casedir=$data_dir/tests schedule=$module", exit_code => 101) };
+
+    like $log, qr/scheduling $module_basename $module\.pm/, 'module scheduled';
+    like $log, qr/Test result \[testresults\/result\-$module_basename\.json\] fail/, 'module test result parsed';
+};
+
+subtest 'exit status from test results: softfailed test' => sub {
+    # dummy isotovideo invocation
+    chdir($pool_dir);
+    path(bmwqemu::STATE_FILE)->remove if -e bmwqemu::STATE_FILE;
+    path('vars.json')->remove if -e 'vars.json';
+    # TODO: backend/baseclass.pm raises if serial0 does not exist.
+    path('serial0')->touch();
+    my $module_basename = 'soft_failed_test';
+    my $module = "tests/$module_basename";
+    my $log = combined_from { isotovideo(
+            default_opts => '--exit-status-from-test-results backend=null',
+            opts => "casedir=$data_dir/tests schedule=$module", exit_code => 0) };
+
+    like $log, qr/scheduling $module_basename $module\.pm/, 'module scheduled';
+    like $log, qr/Test result \[testresults\/result\-$module_basename\.json\] softfail/, 'module test result parsed';
```

Comment:
There is quite some duplication in the three subtests. Can you please extract the common parts so that the actual difference that matters becomes more clear? Also please ensure that the test message strings are unique so that from the output it's already clear where code is executed from.

Reply:
Refactored as a subtest with subtests, it reuses the initial setup.

Reply:
That's nice and looks clean. The test message strings are still ambiguous though

Reply:
Check latest commit.